internal/services/manageddevopspools/managed_devops_pool_resource.go

```diff
@@ -0,0 +1,301 @@
+package manageddevopspools
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/identity"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2025-02-01/projects"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devopsinfrastructure/2025-01-21/pools"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = ManagedDevOpsPoolResource{}
+
+type ManagedDevOpsPoolResource struct{}
+
+func (ManagedDevOpsPoolResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-.]*[a-zA-Z0-9-]$`),
+				"`name` can only include alphanumeric characters, periods (.) and hyphens (-). It must also start with alphanumeric characters and cannot end with periods (.).",
+			),
+		},
+		"resource_group_name":            commonschema.ResourceGroupName(),
+		"location":                       commonschema.Location(),
+		"agent_profile":                  AgentProfileSchema(),
```

Comment:
we do not abstract schema like this, could you please inline it and get rid of this function?

Reply:
Curious to know the reson behind this.
I made it separate, just to avoid a very large resource.go file and keep all schema defn's in one place.
However went ahead and made changes according to your suggesttion.

Do we need to do same for datasource.go as well ?